### ISSUE TITLE
Feature: File Navigation and File Loading

### ISSUE THREAD

**Issue by maxamine** [christopho/solarus-quest-editor#154]
it would be save time if the editor allows to load a file (sprite, tileset, map, script and so) swiftly through an input area, names of files in the current project will be searched, and a preview of the best match will be shown. See screenshots below, it's more explicit:

![1](https://cloud.githubusercontent.com/assets/14140919/14858036/66aa62f2-0c96-11e6-8d74-54f8ccc35e96.png)

![2](https://cloud.githubusercontent.com/assets/14140919/14858041/6bd362ba-0c96-11e6-9ba0-8d1c89057c93.png)

![3](https://cloud.githubusercontent.com/assets/14140919/14858049/71b2e552-0c96-11e6-9737-fe973fca3097.png)

![4](https://cloud.githubusercontent.com/assets/14140919/14857857/a84d1570-0c95-11e6-8294-fbc2f8adbcaf.png)

Add ''file loading'' to the toolbar in the submenu of Edit:

![5](https://cloud.githubusercontent.com/assets/14140919/14857816/828ef844-0c95-11e6-9a07-717ccee97005.png)

Keyboard shortcuts can be used :

**Open File Loading**: Ctrl + L
**Open File Loading**: Esc

Feedback welcome

**Comment by Renkineko** [comment]
"File loading" should be under "File" menu, not "Edit" menu.

**Comment by oclero** [comment]
You mean like in Visual Studio, XCode, Atom and other editors?

**Comment by maxamine** [comment]
the same function is in sublime Text and it greatly facilitates opening files

**Comment by christopho** [comment]
This is a good idea yes.

**Comment by Renkineko** [comment]
Agreed about the sublime text reference, I always use it because it's a time savior if you have some good name standard in your project. As oclero said, the search is really permissive, the search in sublime text is like

between each character, insert a "*" (or a "(.)*" if you want the perl regular expression) and look over all the filepath corresponding with it. So, if you have this project:

/data/sprites/enemies/ganon.dat
/data/sprites/enemies/ganon.png
/data/scripts/sword.lua

And you search "dsrio", you will find all three because of

/**d**ata/**s**p**ri**tes/enemies/gan**o**n.dat
/**d**ata/**s**p**ri**tes/enemies/gan**o**n.png
/**d**ata/**s**c**ri**pts/sw**o**rd.lua

**Comment by maxamine** [comment]
Renkineko: I do not think your proposal would be intuitive, we want the function (Open file by name) facilitates the opening of the files, a good idea to search is to enter the file name (or part of the name) and to ensure that the Editor is looking across all project files with that name regardless of their path. I think it would be more intuitive and easier to open a file.

**Comment by oclero** [comment]
@maxamine have you tried Sublime Text, Atom or Visual Studio Code ?

**Comment by maxamine** [comment]
yes i did but i usually use sublime texte

**Comment by maxamine** [comment]
Renkineko: I don't think your proposal would be intuitive, we want the function (Open file by name) facilitates the opening of files. a good idea to search is to enter the file name (or part of the name) and to ensure that the Editor is looking across all project files with that name regardless of their path (research must also display the file path but without taking it into account in research like you said). I think it would be more intuitive and easier to open a file.

**Comment by Renkineko** [comment]
Not sure about that, you can if you want order the research by files matching the research in first pairs, but if you know you want to open something in a certain folder, you must be able to open by taping directly folder + name. If you clearly highlight why some results is found in the dialog, there will be no confusion possible and it's an advantage if you want to open a file under some folder name without having to search manually with arrow (or worse, with mouse) it's still possible.

I'm talking as the psychorigid coder I am: I always open my files like this in Sublime Text and again, it's a time saver if you know the treeview of your project. I just think Sublime Text should also order results with files matching first then path+file matching, but Sublime Text remember your previous choice so if you put "scri" and opens "/data/scripts/sword.lua" and with "scrip" you open "/data/scripts/shield.lua", it will remember that and highlight the last choice you made with this exact search so the order by doesn't really matter in this context.

**Comment by oclero** [comment]
I agree 100% with Renkineko. Anyways, if you type the name of the file you want to open, it'll be in first position in the list because it matches perfectly. You can also if you want, first type the folder name, then the file name, it will still be in first position.

**Comment by maxamine** [comment]
ok ok understood. If I understood correctly we can search only by file's name and also by path+file names hein?? so it's more global then
